nomad/server.go

```diff
@@ -612,6 +615,30 @@ func (s *Server) State() *state.StateStore {
 	return s.fsm.State()
 }
 
+// Regions returns the known regions in the cluster along with a list
+// of the datacenters within each region.
+func (s *Server) Regions() map[string][]string {
+	s.peerLock.RLock()
+	defer s.peerLock.RUnlock()
+
+	regions := make(map[string][]string, len(s.peers))
+	for region, servers := range s.peers {
```

Comment:
With this implementation, we will get only the regions where the Nomad servers are present. Users might expect the API to return the entire list of regions where the Nomad cluster spans. We might need to change the name of the API or document it to make sure people understand the purpose of the API.